mission_control/src/behaviors/depth_heading.cpp

```diff
@@ -71,22 +72,29 @@ DepthHeadingBehavior::DepthHeadingBehavior(const std::string& name,
   depthHeadingBehaviorPub =
       nodeHandle_.advertise<mission_control::DepthHeading>("/mngr/depth_heading", 100);
 
+  behaviorComplete_ = false;
   behaviorStartTime_ = ros::Time::now();
 }
 
 BT::NodeStatus DepthHeadingBehavior::behaviorRunningProcess()
 {
   if (!goalHasBeenPublished_)
   {
+    behaviorComplete_ = false;
     publishGoalMsg();
+    behaviorStartTime_ = ros::Time::now();
     goalHasBeenPublished_ = true;
   }
   else
   {
     ros::Duration delta_t = ros::Time::now() - behaviorStartTime_;
     if (delta_t.toSec() > timeOut_) setStatus(BT::NodeStatus::FAILURE);
+    else
+    {
+      if (behaviorComplete_) setStatus(BT::NodeStatus::SUCCESS);
+    }
```

Comment:
@patapufate same comment as in elsewhere about stateful actions not allowing re-execution.

Reply:
Fixed